src/ts/client/grapesjs/keymaps.ts

```diff
@@ -0,0 +1,64 @@
+import {Editor, PluginOptions} from 'grapesjs'
+import {cmdPublish} from './PublicationUi'
+import {cmdOpenFonts} from '@silexlabs/grapesjs-fonts'
+import {cmdToggleBlocks, cmdToggleLayers, cmdToggleNotifications, cmdToggleSymbols} from './index'
+import {cmdTogglePages} from './page-panel'
+import {cmdOpenSettings} from './settings'
+
+// Utility functions
+
+const selectBody = (editor: Editor): void => {
+  const wrapper = editor.DomComponents.getWrapper()
+  editor.select(wrapper)
+}
+
+const toggleCommand = (editor: Editor, name: string): void => {
+  const cmd = editor.Commands
+
+  if (!cmd.isActive(name)) {
+    resetPanel(editor)
+    cmd.run(name)
+  } else {
+    cmd.stop(name)
+  }
+}
+
+const resetPanel = (editor: Editor): void => {
+  const panels: string[] = [
+    cmdToggleBlocks,
+    cmdToggleLayers,
+    cmdToggleNotifications,
+    cmdToggleSymbols,
+    cmdTogglePages,
+    cmdOpenSettings,
+    cmdOpenFonts
+  ]
+  panels.forEach(p => editor.Commands.stop(p))
+}
+
+// Main part
+
+export const keymapsPlugin = (editor: Editor, opts: PluginOptions): void => {
+  if (opts.disableKeymaps) return
+
+  const km = editor.Keymaps
+
+  // General keymaps
+  // TODO: Add a keymap to actually publish the site
+
+  // Panels
+  km.add('general:open-settings', 'shift+alt+s', editor => toggleCommand(editor, cmdOpenSettings))
+  km.add('general:open-publish', 'shift+alt+p', editor => toggleCommand(editor, cmdPublish))
+  km.add('general:open-fonts', 'shift+alt+f', editor => toggleCommand(editor, cmdOpenFonts))
+  km.add('panel:preview', 'space', editor => toggleCommand(editor, 'preview'))
+  km.add('panel:layers', 'shift+l', editor => toggleCommand(editor, cmdToggleLayers))
+  km.add('panel:blocks', 'shift+a', editor => toggleCommand(editor, cmdToggleBlocks))
+  km.add('panel:notifications', 'shift+n', editor => toggleCommand(editor, cmdToggleNotifications))
+  km.add('panel:pages', 'shift+p', editor => toggleCommand(editor, cmdTogglePages))
+  km.add('panel:symbols', 'shift+s', editor => toggleCommand(editor, cmdToggleSymbols))
+  km.add('panel:close', 'escape', resetPanel)
```

Comment:
i think it would be better to use tab

and keep space for moving around the canvas like in adobe's software

Reply:
Yes, it makes sense! I was inspired by Blender's way to do it, I believe.

Reply:
All things considered, I don't think it's a good idea to override Tab, since it is used for accessibility purposes to move focus between elements. What do you think?

Reply:
The space bar will definitely be needed at one point to move around

I don't think it makes sense to use silex with accessibility features but maybe i'm wrong

So maybe ctrl+shift+P or something

Reply:
Ok! I think I tried to define "tab" but I couldn't trigger it, will retry. If not, I will add it on Shift+Alt+P, maybe (to stay coherent with the other keymaps, and also because Ctrl+Shift+P triggers the opening of a private browsing window in Firefox).

Reply:
Nevermind, Tab works well!